runbot_gitlab_minimal/controllers/gitlab_ci_controller.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Copyright <2017> <Vauxoo info@vauxoo.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from openerp import http, SUPERUSER_ID
+from openerp.http import request
+from openerp.addons.runbot import runbot
+
+_logger = logging.getLogger(__name__)
+
+
+class RunbotCIController(runbot.RunbotController):
+
+    @http.route(['/runbot/hook_gitlab/<int:repo_id>',
+                 '/runbot/hook_gitlab/org'], type='json', auth="public",
+                website=True)
+    def hook_gitlab(self, repo_id=None, **post):
+        data = request.jsonrequest
+        event = data['object_kind'] if data.has_key('object_kind') else None
+        repository = data['repository']
+        if repo_id is None:
+            if event in ['push', 'merge_request']:
```

Comment:
@moylop260 
I'm not sure if that works because the code is not equal
**This code**
`repo_domain = ['|', ('name', '=', repository['url']),
                                '|',
                                ('name', '=', repository['homepage']),
                                ('name', '=', repository['homepage'] + '.git')]`

**And the parent code**
`repo_domain = [
                    '|', ('name', '=', 'git@github.com:%s/%s.git' % (repo_owner, repo_name)),
                    ('name', '=', 'https://github.com/%s/%s.git' % (repo_owner, repo_name)),
                ]`